### PR TITLE
DOC: Better doc for ``jnp.sum``

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461.

### CODE REVIEW

File: jax/_src/numpy/reductions.py [hunk 1]
```diff
@@ -224,10 +224,83 @@ def _reduce_sum(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None,
...
         out: None = None, keepdims: bool = False, initial: ArrayLike | None = None,
         where: ArrayLike | None = None, promote_integers: bool = True) -> Array:
+  r"""Sum of the elements of the array over a given axis.
+
+  JAX implementation of :func:`numpy.sum`.
+
+  Args:
+    a: Input array.
+    axis: Axis along which the sum to be computed. If axis=None, the sum is
+      computed along both the axes. Default=None.
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:238]: computed along all axes

File: jax/_src/numpy/reductions.py [hunk 2]
```diff
@@ -224,10 +224,83 @@ def _reduce_sum(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None,
...
+    >>> jnp.sum(x, axis=1, keepdims=True)
+    Array([[10],
+           [16],
+           [21]], dtype=int32)
+
+    To include only specific elements in the sum, you can use a``where``.
+
+    >>> where=jnp.array([[False, 0, 1, 0],
+    ...                  [0, 0, 1, 1],
+    ...                  [1, 1, True, 0]])
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:286]: It's strange to mix bool and integer here. Choose one and stick with it

File: jax/_src/numpy/reductions.py [hunk 3]
```diff
@@ -224,10 +224,83 @@ def _reduce_sum(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None,
...
+    >>> jnp.sum(x, axis=1, keepdims=True, where=where)
+    Array([[ 4],
+           [ 9],
+           [12]], dtype=int32)
+    >>> where=jnp.array([[False, 0, 1, 0]])
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:291]: Same here

File: jax/_src/numpy/reductions.py [hunk 4]
```diff
@@ -224,10 +224,83 @@ def _reduce_sum(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None,
...
+    >>> jnp.sum(x, axis=1, keepdims=True, where=where)
+    Array([[4],
+           [6],
+           [3]], dtype=int32)
+
+    If ``dtype`` is specified, ``promote_integers`` will be ignored.
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:297]: I would remove mentions of `promote_integers` from the examples. It's kind of a niche thing that's more of an internal implmentation detail, and the parameter description above seems clear enough on its own.

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/reductions.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Updated accordingly.

File: jax/_src/numpy/reductions.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done.

File: jax/_src/numpy/reductions.py (hunk 4 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Removed.

**Review by jakevdp — APPROVED** [review]
Looks great, thanks!

**Pull request merged**